### PR TITLE
Add token support

### PR DESCRIPTION
**Pull request by travishathaway**
Fixes: #15 

### Important changes

- **Adds support for token based authentication via the `conda_auth.handlers.token` module**
- **Removes authentication verification**; I think this needs to be handled in a separate feature because the current solution I have just doesn't work well for token authentication; this will be handled with #17 
- **Removes the OAuth2 code**; this was just an experimental thing I did when first developing the plugin. Will handle support separately in this issue #7 

### Description

This adds support for token based authentication. It aims to support existing token based authentication with anaconda.org as well as adding support for other "bearer" compatible token authentication platforms (e.g. prefix.dev).

To add a token, run the following command:

```bash
# With an anaconda.org channel
conda auth login <channel> --token <token>
```

```bash
# With a prefix.dev channel
conda auth login https://repo.prefix.dev/<channel> --token <channel>
```

This will trigger a new entry to be written to your `.condarc` file. It looks like this:

```yaml
channel_settings:
- channel: <channel>
  username: token
  auth: token
- channel: https://repo.prefix.dev/<channel>
  username: token
  auth: token
```

On subsequent runs with conda, that configuration will be read and the correct token will be found via the keyring library.

To remove the token from the keyring, run:

```
conda auth logout <channel>
```

### CODE REVIEW

**Comment by wolfv** [comment]
This is great!

File: conda_auth/cli.py [hunk 1]
```diff
@@ -76,58 +113,77 @@ def auth_wrapper(args):
...
 
 
 @group.command("logout")
 @click.argument("channel", callback=parse_channel)
-def logout(channel: Channel):
+@click.pass_obj
+def logout(extra_context: ExtraContext, channel: Channel):
     """
-    Logout of a channel
+    Log out of a by removing any credentials or tokens associated with it.
```
Comment (beeankha) [inline conda_auth/cli.py:179]:
```suggestion
    Log out of a channel by removing any credentials or tokens associated with it.
```

**Comment by beeankha** [comment]
Would it be practical to remove the channel name + auth type from the `.condarc` file after a `logout` command is run? Not a blocker for this by any means, just curious. For example, if there's a slightly mistyped channel name that shows a "success" message with a login attempt, that channel with a typo is persistent in the config file no matter what (at least from my understanding).

**Review by beeankha — APPROVED** [review]

**Comment by travishathaway** [comment]
@beeankha,

I thought about doing this, but `conda auth` isn't the only command that may be editing these settings in the future, so removing a channel completely could be removing whatever other settings are associated with it.

I think for now we should just let users edit this file themselves. A future feature request could be to remove the conda auth related settings for a channel when logging out.

**Pull request merged**